### ISSUE TITLE
Figure out source for uid data

### ISSUE THREAD

**Issue by cc-a** [ImperialCollegeLondon/imperial_coldfront_plugin#20]
Access control for the HPC systems requires the self service system to publish the unix uid data for active users. In order to be able to publish this data we need to be able to get ahold of it so we can store it. The existing self-service platform receives a dump of all active imperial user account data in spreadsheet(?) format that contains the uid and gid info.

This is non-ideal as it is brittle and introduces a lag between user account creation and the self service system being aware of user identities.

Ideally we would pull this data from an appropriate source and store it as a part of the user account create process (in `imperial_coldfront_plugin/oidcs.py`. One known source is the LDAP server `unixldap.cc.ic.ac.uk` though it's not clear how suitable this is as a datasource (with ldap command line utilities `ldapsearch -LLL -H ldaps://unixldap.cc.ic.ac.uk -x -b "ou=everyone,dc=ic,dc=ac,dc=uk" "(cn=ccaveayl)"`).

Is it available from Microsoft Entra?

In short a bit of investigation is required as to the best way to get ahold of this data from College systems.
